blogs/2017/09/28/terminal-renderer.md

```diff
@@ -0,0 +1,96 @@
+---
+Order: 31
+TOCTitle: Integrated Terminal Performance Improvements
+PageTitle: Integrated Terminal Performance Improvements
+MetaDescription: Explore the performance improvements to the integrated terminal renderer in Visual Studio Code
+Date: 2017-09-28
+ShortDescription: Explore the performance improvements to the integrated terminal renderer in Visual Studio Code
+Author: Daniel Imms
+---
+
+# Integrated Terminal Performance Improvements
+
+September 28, 2017 Daniel Imms, [@Tyriar](https://twitter.com/Tyriar)
+
+The rendering engine of the integrated terminal has been completely re-written with performance in mind in the upcoming version 1.17 of VS Code, moving away from a completely DOM-based rendering system to using canvas.
+
+
+
+## DOM Rendering
+
+Somewhat surprisingly, rendering an interactive terminal was totally doable in a system designed for displaying static documents. It was certainly not without its problems though. We found that in order to fix many issues, certain parts of the DOM needed to be overridden:
+
+**Selection**: There was a lot of fighting against the DOM's selection system to cover the terminal use case. Since we always only rendered what was visible to the DOM this meant that you could not select multiple pages of content without reimplementing selection (added in [v1.14](https://code.visualstudio.com/updates/v1_14#_selection-reimplemented))
+
+**Misaligned characters**: Due to many monospace fonts not being strictly monospace for many unicode characters, leading to situations like this:
```

Comment:
Did you have an image you wanted to show here?

Reply:
I do indeed 😄 will update